plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/CompilerSettings.java

```diff
@@ -46,4 +59,20 @@ public boolean getNumericOverflow() {
     public void setNumericOverflow(boolean allow) {
         this.numericOverflow = allow;
     }
+
+    /**
+     * Returns the value for the cumulative total number of statements that can be made in all loops
+     * in a script before an exception is thrown.  This attempts to prevent infinite loops.
```

Comment:
Thanks for this. While I think you might have gone a bit overboard on a few of the comments above this one especially is really useful to have.